_posts/2021-04-18-api-conversion.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title:  "Path to CRD v1"
+date: 2021-04-18
+author: jakub_stejskal
+---
+In Strimzi 0.22 we introduced the `v1beta2` API version to prepare Strimzi for Kubernetes 1.22. The `v1beta2` API version replaces the `v1beta1` and `v1alpha1` API versions.
+This post shows you how to perform an API conversion to `v1beta2` to be ready for the next Strimzi releases.
+
+<!--more-->
+  
+### New API versions
+
+By introducing API version `v1beta2` for Strimzi resources, current Strimzi users must migrate from the old `v1beta1`  and `v1beta2` versions.
+This migration is not necessarily needed right after upgrade to Strimzi 0.22, but has to be done before upgrade to Strimzi 0.23 or later.
+The main reason for the migration is to prepare Strimzi to use [Kubernetes CRD `apiextensions/v1`](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122).
+
+Custom resources are created and controlled using APIs added to Kubernetes by CRDs.
+Put another way, CRDs extend the Kubernetes API to allow the creation of custom resources.
+CRDs are themselves resources within Kubernetes.
+They are installed in a Kubernetes cluster to define the versions of API for the custom resource.
+Each version of the custom resource API can define its own schema for that version.
+Kubernetes clients, including the Strimzi Operators, access the custom resources served by the Kubernetes API server using a URL path (API path), which includes the API version
+
+In 0.22, we still support custom resource version `v1alpha1` for all custom resources and `v1beta1` for `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, `KafkaMirrorMaker`, `KafkaUser`, and `KafkaTopic`.
+However, these versions are currently deprecated and will be dropped in the Strimzi 0.23 release.
+That's one of the reasons why users have to finish the migration before 0.23.
+
+For the new API version, deprecated custom resource properties have been removed. You need to migrate custom resources to the new structure.
+For instance, `v1beta2` only supports a new array-based listener configuration and new metrics configuration formats from config maps.
+`metrics` and `logging` properties have also changed.
+For metrics configuration, you now use `metricsConfig` in `spec.kafka` to reference the name ofa config map with metrics configuration using the `valueFrom` property.
+Same for `logging`, but property name remains same.
+In addition, some old unused fields like `tlsSidecar` were removed as well.
+All the changes to properties are fully described in our [documentation](https://strimzi.io/docs/operators/latest/full/deploying.html#proc-upgrade-cli-tool-files-str).
+You can also see the deprecated property as a warning in the custom resource status
```

Comment:
> You can also see the deprecated property as a warning in the custom resource status

This feels weird. I would say `You can also see a warning in the custom resource status when the deprecated property is used.` or similar.